### PR TITLE
[EOS-20542,EOS-20544,EOS-20554,EOS-20561] : Updates and fixes in Account Login Profile 

### PR DESCRIPTION
**Pull request by shalakadharap**
Update Account Login Profile using LDAP credentials shows inappropriate Auth Logs .

### CODE REVIEW

File: auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java [hunk 1]
```diff
@@ -151,11 +151,16 @@
...
+      LOGGER.error("Account name is either null or empty");
+      return accountResponseGenerator.invalidRequest(
+          "Please provide account name");
+    }
     try {
       account = accountDAO.find(requestBody.get("AccountName"));
       if (!account.exists()) {
-        LOGGER.error("Account [" + requestor.getAccount().getName() +
-                     "] does not exists");
+        LOGGER.error("Requested account does not exists");
```
Comment (bkirunge7) [inline auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java]: log value of account name

File: auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java [hunk 2]
```diff
@@ -182,10 +188,17 @@
...
 
           if (requestBody.get("PasswordResetRequired") != null) {
+            isRequiredInputProvided = true;
             account.setPwdResetRequired(
                 requestBody.get("PasswordResetRequired").toUpperCase());
             LOGGER.info("Updating password reset required flag");
           }
+          if (!isRequiredInputProvided) {
+            LOGGER.error(
+                "Neither password not password-reset flag is provided");
```
Comment (bkirunge7) [inline auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java:199]: what if we add more arguments in future, these log messages will change, so better log as missing mandatory input arguments

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java [hunk 3]
```diff
@@ -193,6 +175,27 @@ ServerResponse serve(FullHttpRequest httpRequest,
...
           return serverResponse;
         }
+
+      } else {
+
+        try {
+          requestor = RequestorService.getRequestor(clientRequestToken);
+        }
+        catch (InvalidAccessKeyException ex) {
+          LOGGER.debug(ex.getServerResponse().getResponseBody());
```
Comment (bkirunge7) [inline auth/server/src/main/java/com/seagates3/controller/IAMController.java:185]: these should be logged in error level

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java [hunk 4]
```diff
@@ -193,6 +175,27 @@ ServerResponse serve(FullHttpRequest httpRequest,
...
+        }
+        if (requestor == null) {
+          LOGGER.error("Invalid user, authentication failed");
+          AuthenticationResponseGenerator responseGenerator =
+              new AuthenticationResponseGenerator();
+          serverResponse = responseGenerator.invalidAccessKey();
+          return serverResponse;
       } else {
         LOGGER.debug("Validating user with accesskey and secretkey entered");
```
Comment (bkirunge7) [inline auth/server/src/main/java/com/seagates3/controller/IAMController.java:200]: remove these log statements may not be use full

File: auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java (hunk 1 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java]: handled

File: auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java (hunk 2 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/AccountLoginProfileController.java:199]: fixed

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java (hunk 3 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/IAMController.java:185]: fixed

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java (hunk 4 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/IAMController.java:200]: removed

**Review by sachitanands — APPROVED** [review]
looks good to me.

**Review by amitwac2608 — APPROVED** [review]

**Comment by shalakadharap** [comment]
Rebuild jenkins

**Review by bkirunge7 — APPROVED** [review]

**Comment by shalakadharap** [comment]
Mini Provisioning success-
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1034/
VM Deployment success-
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/S3Server/602/
Jenkins success-
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/1008/

**Comment by nileshgovande** [comment]
All checks have passed - merging

**Pull request merged**